domain/schema/model/sql/0015-charm.sql

```diff
@@ -126,7 +130,7 @@ INSERT INTO hash_kind VALUES
 
 CREATE TABLE charm_hash (
     charm_uuid TEXT NOT NULL,
-    hash_kind_id TEXT NOT NULL,
+    hash_kind_id INT NOT NULL DEFAULT 0,
```

Comment:
Should this table be populated in this file? Since I guess it's never going to change over the lifetime of juju

Reply:
This is populated via the `setCharmHash` method. It's default to 0, as it's unlikely to change, but we do enforce the setting of the value.